docs/docs.sh

```diff
@@ -0,0 +1,142 @@
+#!/bin/bash
+
+# Get the 1st argument passed to the script and convert it to lowercase
+arg1=$(echo $1 | tr '[:upper:]' '[:lower:]')
+arg2=$(echo $2 | tr '[:upper:]' '[:lower:]')
+
+
+# Build the MkDocs container which will be used to build/serve the MkDocs project
+build_container () {
+    echo "Building the container..."
+    docker build -t mkdocs:latest ./mkdocs/
+    if [ $? -ne 0 ]; then
+        echo "Failed to build the container"
+        exit 1
+    fi
+}
+
+drift_dev () {
+    echo "Running Drift Dev..."
+    dart run drift_dev schema steps lib/snippets/migrations/exported_eschema/ lib/snippets/migrations/schema_versions.dart
+    if [ $? -ne 0 ]; then
+        echo "Failed to run Drift Dev"
+        exit 1
+    fi
+    dart run drift_dev schema generate --data-classes --companions lib/snippets/migrations/exported_eschema/ lib/snippets/migrations/tests/generated_migrations/
+    if [ $? -ne 0 ]; then
+        echo "Failed to run Drift Dev"
+        exit 1
+    fi
+}
+
+serve_mkdocs () {
+    echo "Running MkDocs serve..."
+    docker run --rm -p 9000:9000 -v $(pwd):/docs --user $(id -u):$(id -g) mkdocs:latest serve -f /docs/mkdocs/mkdocs.yml -a 0.0.0.0:9000
+}
+
+if [ $arg1 == "build" ]; then
+
+    echo "Building the project..."
+
+    # Remove the `build` directory if it already exists
+    rm -r ./deploy
+    rm -r ./build
+
+    # Activate the webdev command
+    dart pub global activate webdev
+
+    drift_dev
+
+    # The below command will compile the dart code in `/web` to js & run build_runner
+    webdev build -o web:build/web -- --delete-conflicting-outputs
+    if [ $? -ne 0 ]; then
+        echo "Failed to build the project"
+        exit 1
+    fi
+
+    # Run the build_runner command to generate files in the `test` directory
+    dart run build_runner build --delete-conflicting-outputs
```

Comment:
`webdev build` has an optimization to not run a full build since we only care about `web`. Since we're running a full build afterwards, let's run `build_runner` first and then use `webdev` to emit the files.

Also, let's add a dev dependency on `webdev` and use `dart run webdev` instead of relying on global dependencies.  And to be sure, pass `--release` to both `build_runner build` and `webdev build`.

Reply:
Done